### PR TITLE
Chore/enforce pitr for enabling branching

### PR DESCRIPTION
**Pull request by joshenlim**
If enabling branching when PITR is not disabled, but on Free plan
image

If enabling branching when PITR is not disabled, but on Pro plan already
image

If disabling PITR when branching is enabled:
image

### CODE REVIEW

File: apps/studio/components/interfaces/Settings/Addons/PITRSidePanel.tsx [hunk 1]
```diff
@@ -225,6 +238,21 @@ const PITRSidePanel = () => {
             </div>
           </div>
 
+          {selectedCategory === 'off' && subscriptionPitr !== undefined && isBranchingEnabled && (
+            <Alert_Shadcn_ variant="warning">
+              <AlertTriangleIcon strokeWidth={2} />
+              <AlertTitle_Shadcn_>
+                Removing PITR entails a risk if you run a bad migration from merging your preview
+                branch
```
Comment (kiwicopple) [inline apps/studio/components/interfaces/Settings/Addons/PITRSidePanel.tsx:246]:
```suggestion
                Are you sure you want to disable this while using Branching?
```

File: apps/studio/components/interfaces/Settings/Addons/PITRSidePanel.tsx [hunk 2]
```diff
@@ -225,6 +238,21 @@ const PITRSidePanel = () => {
...
+              </AlertTitle_Shadcn_>
+              <AlertDescription_Shadcn_>
+                Without PITR, you will not be able to restore your project to a healthy state in the
+                possible event of a bad migration when merging your preview branches into your
+                production branch.
```
Comment (kiwicopple) [inline apps/studio/components/interfaces/Settings/Addons/PITRSidePanel.tsx:251]:
```suggestion
                Without PITR, you might not be able to recover lost data if you accidentally
                merge a branch that deletes a column or user data. We don't recommend this.
```

File: apps/studio/components/layouts/AppLayout/EnableBranchingButton/EnableBranchingModal.tsx [hunk 3]
```diff
@@ -134,73 +159,141 @@ const EnableBranchingModal = () => {
...
+                      </figure>
+                    </div>
+                    <div>
+                      <p className="text-sm text-foreground">
+                        You will not be able to use the dashboard to make changes to the database
+                      </p>
+                      <p className="text-sm text-foreground-light">
+                        Schema changes for database preview branches must be done via Git. We are
+                        nonetheless working on allowing the dashboard to make schema changes for
+                        preview branches.
```
Comment (kiwicopple) [inline apps/studio/components/layouts/AppLayout/EnableBranchingButton/EnableBranchingModal.tsx:214]:
```suggestion
                        Schema changes for database Preview Branches must be done using git. 
                        Dashboard changes to Preview Branches are coming soon.
```

File: apps/studio/components/layouts/AppLayout/EnableBranchingButton/EnableBranchingModal.tsx [hunk 4]
```diff
@@ -134,73 +159,141 @@ const EnableBranchingModal = () => {
...
+              <div className="">
+                <Alert_Shadcn_ className="rounded-none border-r-0 border-l-0 px-7 [&>svg]:left-6">
+                  <AlertCircleIcon strokeWidth={2} />
+                  <AlertTitle_Shadcn_>
+                    Point in time recovery (PITR) is required for branching
+                  </AlertTitle_Shadcn_>
+                  <AlertDescription_Shadcn_>
+                    This is to ensure that your project can always be restored to a healthy state in
+                    the event of a bad migration that might occur when merging a preview branch into
+                    your production branch.
```
Comment (kiwicopple) [inline apps/studio/components/layouts/AppLayout/EnableBranchingButton/EnableBranchingModal.tsx:257]:
```suggestion
                    This is to ensure that you can always recover data if you make a "bad migration".
                    For example, if you accidentally delete a column or some of your production data.
```

File: apps/studio/components/interfaces/Settings/Addons/PITRSidePanel.tsx [hunk 5]
```diff
@@ -225,6 +238,19 @@ const PITRSidePanel = () => {
             </div>
           </div>
 
+          {selectedCategory === 'off' && subscriptionPitr !== undefined && isBranchingEnabled && (
+            <Alert_Shadcn_ variant="warning">
+              <AlertTriangleIcon strokeWidth={2} />
+              <AlertTitle_Shadcn_>
+                Are you sure you want to disable this while using Branching?
```
Comment (MildTomato) [inline apps/studio/components/interfaces/Settings/Addons/PITRSidePanel.tsx:245]: came here to suggest something like this :D

**Review by MildTomato — APPROVED** [review]

**Pull request merged**